Rubberduck.VBEEditor/SafeComWrappers/VBA/VBComponents.cs

```diff
@@ -111,34 +111,36 @@ public void ImportSourceFile(string path)
 
             if (ext == ComponentTypeExtensions.DocClassExtension)
             {
+                IVBComponent component;
                 try
                 {
-                    var temp = this[name];
+                    component = this[name];
                 }
                 catch
                 {
                     throw new IndexOutOfRangeException($"Could not find document component named '{name}'.  Try adding a document component with the same name and try again.");
                 }
 
-                var component = this[name];
-                component.CodeModule.Clear();
-
                 var codeString = File.ReadAllText(path, Encoding.UTF8);
-                component.CodeModule.AddFromString(codeString);
+                using (var codeModule = component.CodeModule)
+                {
+                    codeModule.Clear();
+                    codeModule.AddFromString(codeString);
+                }
+                component.Dispose();
```

Comment:
If you can't do a `using` on the `component`, it should be at least in the `finally` block to ensure that exceptions between the successful assignment of `component` up to the `Dispose()` will not leak the variable.

Reply:
Can do that.